gc/2023/GC-05-16.md

```diff
@@ -33,3 +33,176 @@ Installation is required, see the calendar invite.
+
+TL: It would obviously be great for codesize to have this struct.get_indirect opcode because you wouldn't have to generate these dynamic getter  functions for every field of the struct, but it would be interesting to see if there are any performance benefits to
+
+JX: Great suggestion.
+
+BT: If you want to take a struct reference and turn it into the Any type, you need to use the extern.externalize instruction? So you rely on the engine to do the externalization?
+
+JX: Yes, we do rely on the engine or host environment to do externalization.
+
+CW: I imagine for a lot of users of TS, the alternative is to compile to JS, which will be pretty fast. Is the intention to compete with compiling to JS, or just to use TS in an environment where JS isn’t available?
+
+JX: Currently we want to bring more application development languages to Wasm. We don’t expect this approach will be much faster than JS.
+
+CW: Who do you imagine will use this as this first choice?
+
+JX: Developers for LT(???). In the browser, WebAssembly can be some part of the project, if not the whole project. This is more flexible than the current solution.
+
+AR: You mentioned regular typing here but the approach you are using is something close to coercive subtyping and that’s difficult. You definitely need sound subtyping to make it work. But for gradual typing you need to be more flexible, and I’m not sure how that would work with two competing implementations. With graduate typing I should be able to pass an array of C as an array of Any and vice versa. And that could be a mutable array, so you can’t just convert an array to fit the right implementation that you need. You have to keep it in this original presentation, so I’m not sure how you would support that in this approach.
+
+JX: I mentioned the gradual type here, but just want to explain the language. For the static languages, they are trying to support dynamic language features and for dynamic languages, javascript/typescript we are trying to bring more type information to their code. So for gradual typing, wasm might be a good package to bring benefits to these kinds of languages. That does not mean we are going to support all the gradual typing semantics in the languages.
+
+AR: Understood, but how do you even support the most gradual semantics with this approach. In your language, can you assign an array of C to an array of Any?
+
+JX: Not sure what the C to array means.
+
+AR: So C is some static class and you have an array of C, right? And Any basic gradual type should be able to assign that array to a variable of array of type Any. In that case, you cannot wrap the individual objects, it would be expe[snive and violate the identity of the objects because this is a mutable array so you can observe the members changing. How do you support that?
```

Comment:
```suggestion
AR: So C is some static class and you can have an array of C, right? And with basic gradual typing you should be able to assign that array to a variable of array of type Any. In that case, you cannot wrap the individual objects, it would be expensive and in fact, violate the identity of the objects because this is a mutable array so you can observe the members changing. How do you support that?
```